rabbitmq/datadog_checks/rabbitmq/config_models/validators.py

```diff
@@ -11,3 +13,27 @@
 #         raise ValueError('my_number max value is 10, got %s' % str(values.get('my_number')))
 #
 #     return values
+
+
+def initialize_instance(values, **kwargs):
+    if 'prometheus_plugin' in values:
+        plugin_settings = values['prometheus_plugin']
+        if 'url' not in plugin_settings:
+            raise ValueError("prometheus_plugin.url field is required.")
+        if not re.match(r"https?:\/\/", plugin_settings['url']):
+            raise ValueError("prometheus_plugin.url field must be an HTTP or HTTPS URL.")
+        if 'unaggregated_endpoint' in plugin_settings:
+            unagg_ep = plugin_settings['unaggregated_endpoint']
+            if not isinstance(unagg_ep, str):
+                raise TypeError("prometheus_plugin.unaggregated_endpoint must be a string.")
+            if not re.match(r'(per-object|detailed(\?.+)?)$', unagg_ep):
+                raise ValueError(
+                    "prometheus_plugin.unaggregated_endpoint must be either 'per-object' or 'detailed' "
+                    "or 'detailed?<QUERY>'."
+                )
+        if 'include_aggregated_endpoint' in plugin_settings:
+            agg_ep = plugin_settings['include_aggregated_endpoint']
+            if not isinstance(agg_ep, bool):
+                raise TypeError("prometheus_plugin.include_aggregated_endpoint must be a boolean.")
```

Comment:
some of these validations are not necessary, such as type and required, since the default config validators do this already based on the spec.yaml

```suggestion
        if not re.match(r"https?:\/\/", plugin_settings['url']):
            raise ValueError("prometheus_plugin.url field must be an HTTP or HTTPS URL.")
        if 'unaggregated_endpoint' in plugin_settings:
            unagg_ep = plugin_settings['unaggregated_endpoint']
            if not re.match(r'(per-object|detailed(\?.+)?)$', unagg_ep):
                raise ValueError(
                    "prometheus_plugin.unaggregated_endpoint must be either 'per-object' or 'detailed' "
                    "or 'detailed?<QUERY>'."
                )
```
https://github.com/DataDog/integrations-core/blob/4a2aef760394942153081931ceddb8bda4080eaf/rabbitmq/assets/configuration/spec.yaml#L130-L159